src/qt/dc.cpp

```diff
@@ -475,14 +475,17 @@ bool wxQtDCImpl::DoGetPixel(wxCoord x, wxCoord y, wxColour *col) const
     if ( col )
     {
         wxCHECK_MSG( m_qtPixmap != NULL, false, "This DC doesn't support GetPixel()" );
-        QImage image = m_qtPixmap->toImage();
-        QColor pixel = image.pixel( x, y );
+        QPixmap pixmap1px = m_qtPixmap->copy( x, y, 1, 1 );
+        QImage image = pixmap1px.toImage();
+        QColor pixel = image.pixel( 0, 0 );
```

Comment:
A small further optimization would be the use of const references instead of local copies for the 3 local variables above. Again, they can be _const references_ and not plain references.
https://herbsutter.com/2008/01/01/gotw-88-a-candidate-for-the-most-important-const/

Reply:
FWIW, it looks like both `QPixmap` and `QImage` use reference counting and COW (copy-on-write), so copying them is not expensive, but, of course, avoiding the copy by using a const reference to the temporary object (whose lifetime is extended until the end of the containing block), is still faster, even if only marginally.